libs/use-multi-media-devices/src/index.ts

```diff
@@ -0,0 +1,474 @@
+import { useState, useEffect, useReducer } from 'react';
+
+export type MediaDevices = {
+  cameraList: InputDeviceInfo[];
+  microphoneList: InputDeviceInfo[];
+  streams: StreamsMap;
+  initDefaultStream: () => void;
+  addStream: ({
+    type,
+    microphone,
+    camera,
+    audioConstraints,
+    videoConstraints,
+  }: {
+    type: StreamTypes;
+    microphone?: InputDeviceInfo;
+    camera?: InputDeviceInfo;
+    audioConstraints?: MediaTrackConstraints;
+    videoConstraints?: MediaTrackConstraints;
+  }) => Promise<void>;
+  toggleAudio: (id: StreamId) => void;
+  toggleVideo: (id: StreamId) => void;
+  removeStream: (id: StreamId) => void;
+  applyConstraints: ({
+    id,
+    audioConstraints,
+    videoConstraints,
+  }: {
+    id: StreamId;
+    audioConstraints?: MediaTrackConstraints;
+    videoConstraints?: MediaTrackConstraints;
+  }) => Promise<void>;
+  reset: () => void;
+};
+
+export type Resolution = {
+  name?: string;
+  width: number;
+  height: number;
+};
```

Comment:
Not PR-specific, but can I suggest implementing a new pattern where we put types in a separate file? In the interest of minimising per-file LOC, it would be better to define TS types in a colocated `./types.ts` (under the same directory) and import as necessary.

Reply:
We definitely need to use this approach in the future